reana_commons/k8s/kerberos.py

```diff
@@ -115,14 +115,22 @@ def get_kerberos_k8s_config(
                 f"while ! test -f {KRB5_STATUS_FILE_LOCATION}; do "
                 f"if [ $SECONDS -ge {KRB5_TICKET_RENEW_INTERVAL} ]; then "
                 'echo "Renewing Kerberos ticket: $(date)"; kinit -R; SECONDS=0; fi; '
-                f"sleep {KRB5_STATUS_FILE_CHECK_INTERVAL}; done"
+                # wait until status file is created or for a given timeout, whichever comes first
+                f"inotifywait --quiet --format 'Created file %w%f, stopping inotifywait' --timeout {KRB5_STATUS_FILE_CHECK_INTERVAL} --event create {KRB5_TOKEN_CACHE_LOCATION}; "
```

Comment:
Note that this creates the following in the job logs:

```
krb5-renew: :
 Created file /krb5_cache/status_file, stopping inotifywait
```

This phrase is exposed to researchers, and may be hard to understand.

Can we reformulate it to something more user-friendly, such as "Job status change detected, stopping Kerberos token renewal sidecar" perhaps?

Reply:
Normally only the "status file" will be created there, but in general any file that is created inside `KRB5_TOKEN_CACHE_LOCATION` will make `inotifywait` write that phrase and stop. Maybe we should use a phrase like `Job status change detected`, without also saying that the Kerberos sidecar has stopped? We can then add an `echo` command outside the loop to report the end of the execution of the sidecar.

(on this note, newer versions of `inotifywait`  have the `--include` option to only considers event on a subset of files matching a given pattern, but this is not offered by the one present in reana-auth-krb5)

Reply:
Split the phrase in two: first `Detected job status change` is printed, then `Stopping Kerberos ticket renewal sidecar` when exiting the loop